### DISCUSSION TITLE
ux: unexpected behavior after selecting `No` after being asked `Always configure projects upon opening?`

### DISCUSSION THREAD

**Discussion by ymleung314** [microsoft/vscode-cmake-tools#3597] [General]
When I opened a directory with a CMake project, I was asked `Always configure projects upon opening?`, which I clicked `No`. Then, the workspace setting `cmake.configureOnOpen` is set to `true` automatically. However, I don't see why, when I literally just said "don't configure projects upon opening", the currently opened workspace should be configured on open.

**Comment by gcampbell-msft** [comment]
Are you able to reliably reproduce this? If so, this is definitely a bug, and we'd like to have a GitHub issue created for this to track and investigate. Please let us know so that we can investigate. Thanks!

**Reply by ymleung314** [reply]
The extension somehow stopped showing that prompt for me. But by inspecting the source code, there still seems to be the logic error. See [`vscode-cmake-tools/src/extension.ts`](https://github.com/microsoft/vscode-cmake-tools/blob/ec03fc1efbc5e62fb3666b4fc1dde2c1024c4e4c/src/extension.ts) from line 601. If I chose `Yes` as `Would you like to configure project ...?` is asked, `chosen.doConfigure` would be true. Later at line 641, `chosen.doConfigure` which is true would be stored into the configuration even if I chose `No` to `Always configure projects upon opening?`.

**Reply by gcampbell-msft** [reply]
Ah, I see the misunderstanding, though I will admit that it is slightly confusing. 

The goal of these two pop-ups is to answer two different questions. 

The first `Would you like to configure project ...?` answers the question of whether you'd like to configure on open. 
The second `Always configure projects upon opening?` is to understand at what scope you want to set `configureOnOpen`. If you say yes, then we save your answer to `Would you like to configure project ...?` at the global level, if you say no, then we only save it at the workspace level. 

Therefore, if you don't want to configure on open, then you should first answer `No` to `Would you like to configure project ...?`.

**Reply by ymleung314** [reply]
Oh lol. I suggest changing those questions a bit then. They are indeed quite confusing.

**Reply by gcampbell-msft** [reply]
Thanks for the feedback. We've gotten lots of feedback about those pop-ups, both about the wording and the presence of them at all, so we are in the midst of investigating simply changing to a default of `true` for `cmake.configureOnOpen`. Then, users can modify the setting themselves if desired. 

What are your thoughts on this?

**Reply by ymleung314** [reply]
I wonder if you guys have already changed that as I am no longer seeing that prompt, straight up configuration on open every time. I would say, for an existing user, that could feel surprising. That also means trusting the workspace leads to immediate configuration, not something everyone always wants.

**Reply by gcampbell-msft** [reply]
That's good feedback. However, to ease some of those concerns, we would only be changing the default, so anyone who has already answered "yes" or "no" to the configureOnOpen UI, this would not affect. 

The workspace trust is a good callout though.

**Reply by ymleung314** [reply]
So those prompts will never pop up?

**Reply by gcampbell-msft** [reply]
Correct
